model/postgres/postgres_server.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative "../../model"
+
+class PostgresServer < Sequel::Model
+  one_to_one :strand, key: :id
+  many_to_one :resource, class: PostgresResource, key: :resource_id
+  many_to_one :vm
```

Comment:
Yes, but one_to_one expects to see foreign key in the associated table, in this case Vm table. However we put the foreign key to the PostgresServer table.

> [many_to_one](https://github.com/jeremyevans/sequel/blob/master/doc/association_basics.rdoc#many_to_one-)[¶](https://github.com/jeremyevans/sequel/blob/master/doc/association_basics.rdoc#label-many_to_one) [↑](https://github.com/jeremyevans/sequel/blob/master/doc/association_basics.rdoc#top)
The many_to_one association is used when the table for the current class contains a foreign key that references the primary key in the table for the associated class. It is named ‘many_to_one’ because there can be many rows in the current table for each row in the associated table.

Reply:
Yes, we are experiencing the same issue with the GithubRunner model. However, we can override its expectations.

https://github.com/ubicloud/ubicloud/blob/d0bec18866fb5ad8c43e9c4a3f9215182fb834bf/model/github_runner.rb#L8

Reply:
cool, didn't know about the primary_key option